src/content/developers/docs/consensus-mechanisms/pow/mining-algorithms/ethash/index.md

```diff
@@ -0,0 +1,1007 @@
+---
+title: Ethash
+description: A detailed look at the Ethash algorithm.
+lang: en
+sidebar: true
+incomplete: false
+---
+
+[Ethash](https://github.com/ethereum/wiki/wiki/Ethash) is a modified version of the [Dagger-Hashimoto](/developers/docs/consensus-mechanisms/pow/mining-algorithms/dagger-hashamoto) algorithm. Ethash PoW is memory hard, making it ASIC resistant. Memory hardness is achieved with a proof of work algorithm that requires choosing subsets of a fixed resource dependent on the nonce and block header. This resource (a few gigabytes in size) is called a DAG. The DAG is totally different every 30000 blocks, a 125-hour window called an epoch (roughly 5.2 days) and takes a while to generate. Since the DAG only depends on block height, it can be pre-generated but if it's not, the client needs to wait until the end of this process to produce a block. If clients do not pre-generate and cache DAGs ahead of time the network may experience massive block delay on each epoch transition. Note that the DAG does not need to be generated for verifying the PoW essentially allowing for verification with both low CPU and small memory.
```

Comment:
Ethash is not really asic resistant anymore
"totally different"

Reply:
thanks, updated